var/spack/repos/builtin/packages/py-cffi/package.py

```diff
@@ -32,12 +33,22 @@ class PyCffi(PythonPackage):
     # ./spack-src/cffi/ffiplatform.py has _hack_at_distutils which imports
     # setuptools before distutils, but only on Windows. This could be made
     # unconditional to support Python 3.12
-    depends_on("python@:3.11", type=("build", "run"))
+    depends_on("python@:3.11", type=("build", "run"), when="@:1.15")
+    depends_on("python@:3.12", type=("build", "run"), when="@1.16:")
     depends_on("pkgconfig", type="build")
     depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@66.1:", type="build", when="@1.16:")
```

Comment:
```suggestion
    depends_on("py-setuptools@66.1:", type="build", when="@1.17:")
```

Reply:
The pyproject for 0.16 lists @66.1:

Reply:
https://github.com/python-cffi/cffi/blob/release-1.16/pyproject.toml#L5